content/en/_index.md

```diff
@@ -1028,9 +1203,86 @@ def read_a_file file_path:
   puts "Contents: #{content}"
 end
 {{< /highlight >}}
+{{< /content-tab >}}
+{{< content-tab header="Terraform" >}}
+#### No CLIs
+
+Do not include CLI commands (such as gcloud or kubectl) inside
+of your sample via the `null_resource` resource.
+
+#### No Variables
+
+Don't include Terraform variables (`var.VARIABLE_NAME`). Variables
+enabling users to use Terraform samples not as snippets but as direct
+module references. Disabling the avenue of customization 
+reduces the risk so that users don't take accidental 
+dependency.
+Instead, hard code the resource names and attribute values.
+In some cases, [locals](#locals) are allowed.
+
+#### Locals {#locals}
+
+Sometimes multiple resources have a common string that is not
+accessible as a reference or there is no child/parent
+reference. Locals are allowed and encouraged for repeated text
+that a user may want to change. When a name is used multiple
+times, it's easier to copy, paste, and edit a local than search
+and replace the whole file. Locals are allowed and variable blocks
+are not because locals are an internal implementation and not
+part of the API  exposed via Terraform. Locals should be used only
+when necessary, and are best used when a repeated value is used
+throughout the configuration (3 or more times).
+
+#### Outputs
+
+Output statement provide values that are generated after creating
+a resource. They are allowed because they are readonly and they
+give  users some important information such as URLs and IP addresses.
+To keep it simple, they must be defined inline at the end of
+a  sample, not in a separate `outputs.tf` file.
+
+#### Resource references
+
+When possible, refer to already-defined resources.
+
+#### Terraform Google Provider version
+
+When your sample contains a new resource, add the required minimum
+Terraform Google Provider version corresponding to which 
+provider version introduced a resource. For example, `google_apigee_nat_address`
+was added in version 4.37, so the minimum version is 4.37. To specify a minimum
+version in a sample, include the provider requirements, as follows:
+
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+      version = "~= 4.37.0"
+    }
+  }
+}
+
+To determine when a resource was added, see the terraform-provider-google/CHANGELOG.md or 
+terraform-provider-google-beta/CHANGELOG.md.
+
+#### Provider argument
+
+Add the provider argument if the resource is from provider google-beta.
+resource "google_<resource_name>" "default" {
+  provider = google-beta
+  ...
+}
+
+Fields and resources that are only present in google-beta are clearly marked in
+the provider documentation.
+{{< /content-tab >}}
+
+{{< /content-tabpane >}}
 
 ### Testing practices
 
+{{< content-tabpane >}}
```

Comment:
blocker: We are in a section of the style guide where all the guideilnes are meant to be inside language-specific tab panes. For this testing practices heading to be excluded from Terraform and Ruby here means something is not organized correctly. Previewing would probably surface the problem quickly.

#StyleGuideFeedback

Reply:
fixed